hs_core/views/__init__.py

```diff
@@ -207,6 +235,8 @@ def add_files_to_resource(request, shortkey, *args, **kwargs):
         msg = 'validation_error: ' + ex.message
         return HttpResponse(msg, status=500)
 
+    web_services.update(shortkey)
+
```

Comment:
Need to put web_services.update() call into a celery task so it runs in a background thread so as not to block the main thread, especially given the calling functions are called very frequently.